holoviews/core/spaces.py

```diff
@@ -551,6 +551,21 @@ def _cross_product(self, tuple_key, cache):
         return self.clone(data)
 
 
+    def _slice_bounded(self, tuple_key):
+        """
+        Slices bounded DynamicMaps by setting the soft_ranges on key dimensions.
+        """
+        cloned = self.clone(self)
+        for i, slc in enumerate(tuple_key):
+            (start, stop) = slc.start, slc.stop
+            if start and start < cloned.kdims[i].range[0]:
+                raise Exception("Requested slice below defined dimension range.")
+            if stop and stop > cloned.kdims[i].range[1]:
```

Comment:
Make sure you check ``start is not None`` and ``stop is not None`` explicitly unless it'll fall through for zeros.